Duplicati/Library/Main/Operation/BackupHandler.cs

```diff
@@ -103,8 +103,18 @@ public static Snapshots.ISnapshotService GetSnapshot(string[] sources, Options o
         private UsnJournalService GetJournalService(IEnumerable<string> sources, ISnapshotService snapshot, IFilter filter, long lastfilesetid)
         {
             if (m_options.UsnStrategy == Options.OptimizationStrategy.Off) return null;
+
+            // get hash identifying current source filter / sources configuration
+            // ReSharper disable once PossibleMultipleEnumeration
+            var configHash = Library.Utility.Utility.ByteArrayAsHexString(MD5HashHelper.GetHash(
+                (filter == null ? string.Empty : filter.ToString()) +
+                string.Join("; ", sources) +
+                m_options.FileAttributeFilter.ToString() +
+                m_options.SkipFilesLargerThan.ToString()
+            ));
+
             var journalData = m_database.GetChangeJournalData(lastfilesetid);
-            var service = new UsnJournalService(sources, snapshot, filter, journalData, cancellationTokenSource.Token);
+            var service = new UsnJournalService(sources, snapshot, filter, configHash, journalData, cancellationTokenSource.Token);

```

Comment:
Is there any downside to passing in `m_options` instead?  This would keep the `UsnJournalService` responsible for the hash calculation and would make it a bit easier to create instances of `UsnJournalService` elsewhere in the future if needed.

Reply:
That sounds like a good idea .. I'll adjust. Thanks!

Reply:
@warwickmm - actually, I'm not sure how to accomplish this. I cannot reference the Options class directly in UsnJournalService.cs without adding a reference to Duplicati.Library.Main to Duplicati.Library.Snapshots.  But this causes a circular reference so is not allowed. I'm not super fluent in C# so maybe there's a simple solution I'm overlooking.

Reply:
I see, that's unfortunate.  Perhaps that's why many classes take a `Dictionary<string, string>` of options, which I would like to eliminate as well.
  
What are your thoughts about simply adding a `FileAttributesFilter fileAttributesFilter` and a `long maxFileSizeFilter` argument to the `UsnJournalService` constructor?  The way you've written this is fine, but it just feels a bit nicer to keep the hash calculation inside the `UsnJournalService` class.

It might make sense to later introduce an object that encapsulates all the possible exclusions.  This would make it easier to manage in the future, especially if we start to introduce usage of Linux `fanotify` filesystem events.

Reply:
> What are your thoughts about simply adding a FileAttributesFilter fileAttributesFilter and a long maxFileSizeFilter argument to the UsnJournalService constructor?

Yep, I can do that.  It's how I was originally going to implement this, but then thought it'd be cleaner this way.  But I agree with your thoughts on keeping the hash calculation in UsnJournalService.  I'll make the change soon.